In [2]:
import time
from datetime import datetime, date
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
pd.set_option('display.notebook_repr_html',True)
pd.set_option('display.max_column',7)
pd.set_option('display.max_rows',10)
pd.set_option('display.width',250)

In [3]:
dates = pd.date_range('2021-07-01','2021-07-06')
print(dates)
print(type(dates))

DatetimeIndex(['2021-07-01', '2021-07-02', '2021-07-03', '2021-07-04', '2021-07-05', '2021-07-06'], dtype='datetime64[ns]', freq='D')
<class 'pandas.core.indexes.datetimes.DatetimeIndex'>


In [4]:
temps1 = pd.Series([80,82,85,90,83,87], index=dates)
print(temps1)

2021-07-01    80
2021-07-02    82
2021-07-03    85
2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, dtype: int64


In [5]:
temps2 = pd.Series([70,75,69,83,79,77], index=dates)
print(temps2)
temps_diff = temps1-temps2
print(temps_diff)

2021-07-01    70
2021-07-02    75
2021-07-03    69
2021-07-04    83
2021-07-05    79
2021-07-06    77
Freq: D, dtype: int64
2021-07-01    10
2021-07-02     7
2021-07-03    16
2021-07-04     7
2021-07-05     4
2021-07-06    10
Freq: D, dtype: int64


In [6]:
print(temps_diff.mean())
print(type(temps_diff))

9.0
<class 'pandas.core.series.Series'>


In [7]:
temps_df = pd.DataFrame({'Missoula':temps1, 'Philadelphia':temps2})
print(temps_df)

            Missoula  Philadelphia
2021-07-01        80            70
2021-07-02        82            75
2021-07-03        85            69
2021-07-04        90            83
2021-07-05        83            79
2021-07-06        87            77


In [8]:
# 실습-3-해답 

temps_1 = temps1 + temps2
temps_2 = temps1 * temps2
temps_3 = temps1 / temps2
#
print(temps_1)
print(temps_2)
print(temps_3)

2021-07-01    150
2021-07-02    157
2021-07-03    154
2021-07-04    173
2021-07-05    162
2021-07-06    164
Freq: D, dtype: int64
2021-07-01    5600
2021-07-02    6150
2021-07-03    5865
2021-07-04    7470
2021-07-05    6557
2021-07-06    6699
Freq: D, dtype: int64
2021-07-01    1.142857
2021-07-02    1.093333
2021-07-03    1.231884
2021-07-04    1.084337
2021-07-05    1.050633
2021-07-06    1.129870
Freq: D, dtype: float64


In [9]:
# 실습-4-해답 
temps3 = pd.Series([75, 77, 67, 82, 77, 85],index=dates)
print('-'*40)
print('temps3')
print(temps3)
dic2 = {'Missoula': temps1,'Philadelphia': temps2,'temps3':temps3}  # 딕셔너리 dic2를 생성함 
temps_df2 = pd.DataFrame(dic2) # 데이터프레임을 생성할 때 인수로 딕셔너리인 dic2를 사용함
print('-'*40)
print('temps_df2')
print(temps_df2)

----------------------------------------
temps3
2021-07-01    75
2021-07-02    77
2021-07-03    67
2021-07-04    82
2021-07-05    77
2021-07-06    85
Freq: D, dtype: int64
----------------------------------------
temps_df2
            Missoula  Philadelphia  temps3
2021-07-01        80            70      75
2021-07-02        82            75      77
2021-07-03        85            69      67
2021-07-04        90            83      82
2021-07-05        83            79      77
2021-07-06        87            77      85


In [10]:
# 실습-5-해답-ver1
print('<<<<실습-1-해답-ver1>>>>')
temps3 = pd.Series([75, 77, 67, 82, 77, 85], index=dates)
temps_df2 = pd.DataFrame(
        {
        'Missoula': temps1,
        'Philadelphia': temps2,
        'Seoul': temps3,
        }
)
print(temps_df2)
print()
print('-'*40)
print('Missoula \t평균기온 : {0:.2f}'.format(temps1.mean()))
print('Philadelphia \t평균기온 : {0:.2f}'.format(temps2.mean()))
print('Seoul \t \t평균기온 : {0:.2f}'.format(temps3.mean()))

<<<<실습-1-해답-ver1>>>>
            Missoula  Philadelphia  Seoul
2021-07-01        80            70     75
2021-07-02        82            75     77
2021-07-03        85            69     67
2021-07-04        90            83     82
2021-07-05        83            79     77
2021-07-06        87            77     85

----------------------------------------
Missoula 	평균기온 : 84.50
Philadelphia 	평균기온 : 75.50
Seoul 	 	평균기온 : 77.17


In [11]:
# 실습-5-해답-ver2
temps3 = pd.Series([75, 77, 67, 82, 77, 85], index=dates)
temps_df2 = pd.DataFrame(
        {
        'Missoula': temps1,
        'Philadelphia': temps2,
        'Seoul': temps3,
        }
)
print()
print('<<<<실습-1-해답-ver2>>>>')
print(temps_df2)
print()
print('-'*40)
#print(      round(temps_df2.mean(),2)      ) # 서비스 코드 주석을 풀어보세요
#print('-'*40)
#print(temps_df2.columns)  # 서비스 코드 주석을 풀어보세요-> 이후 하단의 소스코드와 비교하세요.
print(temps_df2.columns.values[0] ,'\t평균기온 : {0:.2f}'    .format(round(temps_df2.Missoula.mean(),2))     )
print(temps_df2.columns.values[1] ,'\t평균기온 : {0:.2f}'    .format(round(temps_df2.Philadelphia.mean(),2))      )
print(temps_df2.columns.values[2] ,'\t\t평균기온 : {0:.2f}'    .format(round(temps_df2.Seoul.mean(),2))      )


<<<<실습-1-해답-ver2>>>>
            Missoula  Philadelphia  Seoul
2021-07-01        80            70     75
2021-07-02        82            75     77
2021-07-03        85            69     67
2021-07-04        90            83     82
2021-07-05        83            79     77
2021-07-06        87            77     85

----------------------------------------
Missoula 	평균기온 : 84.50
Philadelphia 	평균기온 : 75.50
Seoul 		평균기온 : 77.17


##### 이하는 13일차 학습내용입니다.
##### 이하는 13일차 학습내용입니다.
##### 이하는 13일차 학습내용입니다.

In [12]:
# Pandas DataFrame의 개념 
## 판다스의 시리즈는 각 인덱스 레이블 당 한 개의 값(values)만 가진다.
## 그러나 데이터 프레임을 사용하면 인덱스 한 개당 복수개의 값을 가질 수 있다 
## 데이터 프레임은 인덱스 레이블을 기준으로 배열된 한 개 이상의 시리즈의 집합을 의미한다.
## 각 시리즈는 데이터 프레임의 컬럼(열)이며 각각의 컬럼(열)은 자신만의 이름을 갖는다.
# Pandas DataFrame의 특징 
## 데이터 프레임은 데이터 베이스의 개념과 비슷하다.
## 서로다른 유형의 데이터 컬럼을 한 개 이상 갖는다는 점과, 한 컬럼의 모든 데이터는 동일한 유형이라는 점이 비슷한 점이다.  

In [13]:
## 아래는 두 개의 컬럼을 갖는 DataFrame 객체를 만드는 소스코드이다 
## 각 컬럼은 기온을 나타내는 Series 객체이다
temp_df = pd.DataFrame(   {'Missoula':temps1, 'Philadelphia':temps2}    )
temps_df

Missoula  Philadelphia
2021-07-01        80            70
2021-07-02        82            75
2021-07-03        85            69
2021-07-04        90            83
2021-07-05        83            79
2021-07-06        87            77

In [14]:
print(temps_df['Missoula']) # 데이터 프레임에서 특정한 컬럼을 참조하기
print(type(temps_df))       # temps_df는 데이터프레임 형식임
print(type(temps1))
print(type(temps_df['Missoula']))
aa = temps_df['Missoula']
print(type(aa))

2021-07-01    80
2021-07-02    82
2021-07-03    85
2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, Name: Missoula, dtype: int64
<class 'pandas.core.frame.DataFrame'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>


In [15]:
bb = temps_df.Missoula # 데이터 프레임에 속해 있는 시리즈를 꺼내오는 새로운 문법임
print(bb)                # 직전의 temps_df['Missoula'] 문법과 비교할것
print(type(bb))

2021-07-01    80
2021-07-02    82
2021-07-03    85
2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, Name: Missoula, dtype: int64
<class 'pandas.core.series.Series'>


In [16]:
temps_df['Philadelphia']

2021-07-01    70
2021-07-02    75
2021-07-03    69
2021-07-04    83
2021-07-05    79
2021-07-06    77
Freq: D, Name: Philadelphia, dtype: int64

In [17]:
temps_df.Philadelphia

2021-07-01    70
2021-07-02    75
2021-07-03    69
2021-07-04    83
2021-07-05    79
2021-07-06    77
Freq: D, Name: Philadelphia, dtype: int64

In [18]:
temps_df[['Philadelphia','Missoula']] # 두 개 이상의 컬럼(시리즈)을 한번에 가져올 때에는 대괄호를 중첩하여야 함

Philadelphia  Missoula
2021-07-01            70        80
2021-07-02            75        82
2021-07-03            69        85
2021-07-04            83        90
2021-07-05            79        83
2021-07-06            77        87

In [19]:
temps_diff = temps_df.Missoula - temps_df.Philadelphia # 데이터 프레임에 속한 시리즈 간의 산술연산
print(temps_diff)
print(type(temps_diff))
print(temps_df) # 6번 셀에서 생성되었음 

2021-07-01    10
2021-07-02     7
2021-07-03    16
2021-07-04     7
2021-07-05     4
2021-07-06    10
Freq: D, dtype: int64
<class 'pandas.core.series.Series'>
            Missoula  Philadelphia
2021-07-01        80            70
2021-07-02        82            75
2021-07-03        85            69
2021-07-04        90            83
2021-07-05        83            79
2021-07-06        87            77


In [20]:
temps_df['Difference'] = temps_diff # temps_df라는 데이터 프레임에 temps_diff의 시리즈를 한 개 추가함
                                    # 그리고 df 안에서 추가한 시리즈의 레이블을 'Difference' 라고 정함 
print(temps_df)

            Missoula  Philadelphia  Difference
2021-07-01        80            70          10
2021-07-02        82            75           7
2021-07-03        85            69          16
2021-07-04        90            83           7
2021-07-05        83            79           4
2021-07-06        87            77          10


##  

In [21]:
print(temps_df.columns) # 데이터프레임의 컬럼명을 반환한다.
print(list(temps_df.columns))

Index(['Missoula', 'Philadelphia', 'Difference'], dtype='object')
['Missoula', 'Philadelphia', 'Difference']


In [22]:
# 컬럼명(시리즈 레이블명)변경 
col_list = list(temps_df.columns)
col_list[0] = 'Misu'
temps_df.columns = col_list
print(list(temps_df.columns))

['Misu', 'Philadelphia', 'Difference']


In [23]:
print(temps_df.Misu)

2021-07-01    80
2021-07-02    82
2021-07-03    85
2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, Name: Misu, dtype: int64


In [24]:
# 시리즈 슬라이싱
print(temps_df.Misu[1:4]) # [1:4]슬라이싱
print(temps_df.Misu['2021-07-01':'2021-07-03'])

2021-07-02    82
2021-07-03    85
2021-07-04    90
Freq: D, Name: Misu, dtype: int64
2021-07-01    80
2021-07-02    82
2021-07-03    85
Freq: D, Name: Misu, dtype: int64


In [25]:
print(temps_df.Misu[3:])  # [3:]슬라이싱

2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, Name: Misu, dtype: int64


#### 전체 컬럼에 있어서의 특정 로우 데이터를 보고 싶다면 

#### <code>**.loc**</code>이나 <code>**.iloc**</code> 속성을 사용하면 된다. 
#### <code>**.loc**</code>는 인덱스 레이블을 사용해 검색하며, 
#### <code>**.iloc**</code>는 제로베이스 위치를 사용해 검색한다.

#### 다음은 데이터 프레임 안의 모든 시리즈 중에서 특정 번째의 로우를 가져오는 코드다.

In [26]:
print(temps_df)
print('-'*40)
print(temps_df.iloc[0]) # 이 개념은 데이터 프레임에서 제일 첫 행을 참조하라는 뜻임

            Misu  Philadelphia  Difference
2021-07-01    80            70          10
2021-07-02    82            75           7
2021-07-03    85            69          16
2021-07-04    90            83           7
2021-07-05    83            79           4
2021-07-06    87            77          10
----------------------------------------
Misu            80
Philadelphia    70
Difference      10
Name: 2021-07-01 00:00:00, dtype: int64


In [27]:
print(temps_df.iloc[0].index)

Index(['Misu', 'Philadelphia', 'Difference'], dtype='object')


In [28]:
print(temps_df.loc['2021-07-06'])

Misu            87
Philadelphia    77
Difference      10
Name: 2021-07-06 00:00:00, dtype: int64


In [29]:
print(temps_df.loc['2021-07-01']) # 사용자 인덱스를 참조하는 loc명령은 존재하지 안을 떄 에러 발생함

Misu            80
Philadelphia    70
Difference      10
Name: 2021-07-01 00:00:00, dtype: int64


In [30]:
#013일차-실습-3해답
col_list = list(temps_df.columns)
col_list[0] = 'Missoula'       # 컬럼명(시리즈 레이블명)변경 
temps_df.columns = col_list
#
temps3 = pd.Series([75, 77, 67, 82, 77, 85], index=dates)
temps_df['difference'] = temps1 - temps2 # 
temps_df['Seoul'] = temps3
temps_df

Missoula  Philadelphia  Difference  difference  Seoul
2021-07-01        80            70          10          10     75
2021-07-02        82            75           7           7     77
2021-07-03        85            69          16          16     67
2021-07-04        90            83           7           7     82
2021-07-05        83            79           4           4     77
2021-07-06        87            77          10          10     85

In [31]:
#013일차-실습-4해답
print('-'*40)
print(temps_df.Seoul[-3:])
print('-'*40)
print(temps_df.Seoul[3:6])
print('-'*40)
print(temps_df[['Seoul']][3:])
print('-'*40)
print(temps_df2['Seoul'].tail(3))
print('-'*40)
print(temps_df.loc['20210704':'20210706',['Seoul']])
print('-'*40)
print(temps_df.loc['2021-07-04':'2021-07-06',['Seoul']]) # 위의 소스와 비교

----------------------------------------
2021-07-04    82
2021-07-05    77
2021-07-06    85
Freq: D, Name: Seoul, dtype: int64
----------------------------------------
2021-07-04    82
2021-07-05    77
2021-07-06    85
Freq: D, Name: Seoul, dtype: int64
----------------------------------------
            Seoul
2021-07-04     82
2021-07-05     77
2021-07-06     85
----------------------------------------
2021-07-04    82
2021-07-05    77
2021-07-06    85
Freq: D, Name: Seoul, dtype: int64
----------------------------------------
            Seoul
2021-07-04     82
2021-07-05     77
2021-07-06     85
----------------------------------------
            Seoul
2021-07-04     82
2021-07-05     77
2021-07-06     85


In [32]:
#013일차-실습-5-해답-ver1
print('-'*10,'013일차-실습-5-해답-ver1','-'*10)
print(temps_df.Seoul[temps_df.Seoul >= 80])
#013일차-실습-5-해답-ver2
print('-'*10,'013일차-실습-5-해답-ver2','-'*10)
print(temps_df[temps_df.Seoul >= 80].Seoul)


---------- 013일차-실습-5-해답-ver1 ----------
2021-07-04    82
2021-07-06    85
Name: Seoul, dtype: int64
---------- 013일차-실습-5-해답-ver2 ----------
2021-07-04    82
2021-07-06    85
Freq: 2D, Name: Seoul, dtype: int64


In [33]:
#013일차-실습-6-해답
# 컬럼명(시리즈 레이블명)변경 
col_list = list(temps_df.columns)
col_list[0] = 'Misu'
col_list[1] = 'Phil'
col_list[2] = 'Diff'
col_list[3] = 'diff'
col_list[4] = 'Seou'
temps_df.columns = col_list
temps_df

Misu  Phil  Diff  diff  Seou
2021-07-01    80    70    10    10    75
2021-07-02    82    75     7     7    77
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85

In [34]:
#013일차-실습-6-해답-ver2
temps_df.columns = ["Misu", "Phil", "Diff", "diff", "Seou"]
temps_df

Misu  Phil  Diff  diff  Seou
2021-07-01    80    70    10    10    75
2021-07-02    82    75     7     7    77
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85

##### 013일차-강의-2-시작

In [35]:
print(temps_df.iloc[[ 1,3,5 ]]) # 복수의 아이템 조회시 대괄호 겹쳐쓰기 
                    # temps_df 데이터프레임에서 1,3,5번 인덱스 번호를 가져옴

            Misu  Phil  Diff  diff  Seou
2021-07-02    82    75     7     7    77
2021-07-04    90    83     7     7    82
2021-07-06    87    77    10    10    85


In [36]:
print(temps_df.iloc[[1, 3, 5]].Diff )# 복수의 아이템 조회시 대괄호 겹쳐쓰기 
                                    # 이전의 결과에서 또다시 Diff 열(Diff씨리즈) 에 해당하는 것만 가져옴

2021-07-02     7
2021-07-04     7
2021-07-06    10
Freq: 2D, Name: Diff, dtype: int64


In [37]:
print(temps_df)
print(temps_df.Misu > 82)

            Misu  Phil  Diff  diff  Seou
2021-07-01    80    70    10    10    75
2021-07-02    82    75     7     7    77
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85
2021-07-01    False
2021-07-02    False
2021-07-03     True
2021-07-04     True
2021-07-05     True
2021-07-06     True
Freq: D, Name: Misu, dtype: bool


In [38]:
print(temps_df)
print('='*40)
print(temps_df[temps_df.Misu > 82].Misu)

            Misu  Phil  Diff  diff  Seou
2021-07-01    80    70    10    10    75
2021-07-02    82    75     7     7    77
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85
2021-07-03    85
2021-07-04    90
2021-07-05    83
2021-07-06    87
Freq: D, Name: Misu, dtype: int64


In [39]:
print(temps_df)
print('-'*40)
print(temps_df[temps_df.Misu > 82])

            Misu  Phil  Diff  diff  Seou
2021-07-01    80    70    10    10    75
2021-07-02    82    75     7     7    77
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85
----------------------------------------
            Misu  Phil  Diff  diff  Seou
2021-07-03    85    69    16    16    67
2021-07-04    90    83     7     7    82
2021-07-05    83    79     4     4    77
2021-07-06    87    77    10    10    85


In [40]:
%pfile data/goog.csv

Object `data/goog.csv` not found.


Date,Open,High,Low,Close,Volume
12/19/2016,790.219971,797.659973,786.27002,794.200012,1225900
12/20/2016,796.76001,798.650024,793.27002,796.419983,925100
12/21/2016,795.840027,796.676025,787.099976,794.559998,1208700
12/22/2016,792.359985,793.320007,788.580017,791.26001,969100
12/23/2016,790.900024,792.73999,787.280029,789.909973,623400
12/27/2016,790.679993,797.859985,787.656982,791.549988,789100
12/28/2016,793.700012,794.22998,783.200012,785.049988,1132700
12/29/2016,783.330017,785.929993,778.919983,782.789978,742200
12/30/2016,782.75,782.780029,770.409973,771.820007,1760200
1/3/2017,778.809998,789.630005,775.799988,786.140015,1643100
1/4/2017,788.359985,791.340027,783.159973,786.900024,1065400
1/5/2017,786.080017,794.47998,785.02002,794.02002,1315400
1/6/2017,795.26001,807.900024,792.203979,806.150024,1620500
1/9/2017,806.400024,809.966003,802.830017,806.650024,1272400
1/10/2017,807.859985,809.130005,803.51001,804.789978,1174300
1/11/2017,805,808.150024,801.369995,807.909973,1057900

In [41]:
df = pd.read_csv('data/goog.csv')
print(type(df))
print(df)

<class 'pandas.core.frame.DataFrame'>
          Date        Open        High         Low       Close   Volume
0   12/19/2016  790.219971  797.659973  786.270020  794.200012  1225900
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
2   12/21/2016  795.840027  796.676025  787.099976  794.559998  1208700
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
4   12/23/2016  790.900024  792.739990  787.280029  789.909973   623400
..         ...         ...         ...         ...         ...      ...
56   3/13/2017  844.000000  848.684998  843.250000  845.539978  1149500
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
58   3/15/2017  847.590027  848.630005  840.770020  847.200012  1379600
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400
60   3/17/2017  851.609985  853.400024  847.109985  852.119995  1712300

[61 rows x 6 columns]


In [42]:
df.Date

0     12/19/2016
1     12/20/2016
2     12/21/2016
3     12/22/2016
4     12/23/2016
         ...    
56     3/13/2017
57     3/14/2017
58     3/15/2017
59     3/16/2017
60     3/17/2017
Name: Date, Length: 61, dtype: object

In [43]:
df.Open

0     790.219971
1     796.760010
2     795.840027
3     792.359985
4     790.900024
         ...    
56    844.000000
57    843.640015
58    847.590027
59    849.030029
60    851.609985
Name: Open, Length: 61, dtype: float64

In [44]:
df3 = df.iloc[0:3]
print(df3)

         Date        Open        High         Low       Close   Volume
0  12/19/2016  790.219971  797.659973  786.270020  794.200012  1225900
1  12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
2  12/21/2016  795.840027  796.676025  787.099976  794.559998  1208700


In [45]:
df3 = df.iloc[0:3,4:]
print(df3)

        Close   Volume
0  794.200012  1225900
1  796.419983   925100
2  794.559998  1208700


In [46]:
df4 = df.iloc[0:3, 3:5]
df4

Low       Close
0  786.270020  794.200012
1  793.270020  796.419983
2  787.099976  794.559998

In [47]:
df5 = df.iloc[1:61:2]
df5

Date        Open        High         Low       Close   Volume
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
5   12/27/2016  790.679993  797.859985  787.656982  791.549988   789100
7   12/29/2016  783.330017  785.929993  778.919983  782.789978   742200
9     1/3/2017  778.809998  789.630005  775.799988  786.140015  1643100
..         ...         ...         ...         ...         ...      ...
51    3/6/2017  826.950012  828.880005  822.400024  827.780029  1105800
53    3/8/2017  833.510010  838.150024  831.789978  835.369995   988700
55   3/10/2017  843.280029  844.909973  839.500000  843.250000  1701100
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400

[30 rows x 6 columns]

013일차-실습-7
------------------------------------------------------------------
<문제>, <배경설명>, <요구사항>, <실행결과> 를 충족하는 실습을 수행하고 결과를 제출하세요.
------------------------------------------------------------------
<문제>
df에 있는 자료 중에서 홀수의 행만을 출력하는 코드를 작성하세요.

------------------------------------------------------------------
<요구사항1>
강의-2에서 사용한 파일을 연속하여 실습하세요.

------------------------------------------------------------------
<요구사항2>
- df.iloc[1,3,5,7,9,11,13,15,17, ...] 과 같은 코드는 안됩니다.
- 예제에서 사용한 것과 동일한 코드는 사양합니다

------------------------------------------------------------------
<실행결과>

	Date	Open	High	Low	Close	Volume
1	12/20/2016	796.760010	798.650024	793.270020	796.419983	925100
3	12/22/2016	792.359985	793.320007	788.580017	791.260010	969100
5	12/27/2016	790.679993	797.859985	787.656982	791.549988	789100
7	12/29/2016	783.330017	785.929993	778.919983	782.789978	742200
9	1/3/2017	778.809998	789.630005	775.799988	786.140015	1643100
...	...	...	...	...	...	...
51	3/6/2017	826.950012	828.880005	822.400024	827.780029	1105800
53	3/8/2017	833.510010	838.150024	831.789978	835.369995	988700
55	3/10/2017	843.280029	844.909973	839.500000	843.250000	1701100
57	3/14/2017	843.640015	847.239990	840.799988	845.619995	779900
59	3/16/2017	849.030029	850.849976	846.130005	848.780029	970400
30 rows × 6 columns

------------------------------------------------------------------
이상의 실습결과를 techmr@daum.net; ty@doublerock.io 메일로 전송하세요.
메일제목은 '013일차-실습-7-완료' 라고 지정하세요.

In [48]:
df[df.index % 2 == 1]

Date        Open        High         Low       Close   Volume
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
5   12/27/2016  790.679993  797.859985  787.656982  791.549988   789100
7   12/29/2016  783.330017  785.929993  778.919983  782.789978   742200
9     1/3/2017  778.809998  789.630005  775.799988  786.140015  1643100
..         ...         ...         ...         ...         ...      ...
51    3/6/2017  826.950012  828.880005  822.400024  827.780029  1105800
53    3/8/2017  833.510010  838.150024  831.789978  835.369995   988700
55   3/10/2017  843.280029  844.909973  839.500000  843.250000  1701100
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400

[30 rows x 6 columns]

In [49]:
#013일차-실습-7-해답-ver1
df6 = df.iloc[1::2]
df6

Date        Open        High         Low       Close   Volume
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
5   12/27/2016  790.679993  797.859985  787.656982  791.549988   789100
7   12/29/2016  783.330017  785.929993  778.919983  782.789978   742200
9     1/3/2017  778.809998  789.630005  775.799988  786.140015  1643100
..         ...         ...         ...         ...         ...      ...
51    3/6/2017  826.950012  828.880005  822.400024  827.780029  1105800
53    3/8/2017  833.510010  838.150024  831.789978  835.369995   988700
55   3/10/2017  843.280029  844.909973  839.500000  843.250000  1701100
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400

[30 rows x 6 columns]

In [50]:
#013일차-실습-7-해답-ver2
df6 = df[df.index % 2 == 1]
df6  

Date        Open        High         Low       Close   Volume
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
5   12/27/2016  790.679993  797.859985  787.656982  791.549988   789100
7   12/29/2016  783.330017  785.929993  778.919983  782.789978   742200
9     1/3/2017  778.809998  789.630005  775.799988  786.140015  1643100
..         ...         ...         ...         ...         ...      ...
51    3/6/2017  826.950012  828.880005  822.400024  827.780029  1105800
53    3/8/2017  833.510010  838.150024  831.789978  835.369995   988700
55   3/10/2017  843.280029  844.909973  839.500000  843.250000  1701100
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400

[30 rows x 6 columns]

013일차-실습-8
------------------------------------------------------------------
<문제>, <배경설명>, <요구사항>, <실행결과> 를 충족하는 실습을 수행하고 결과를 제출하세요.
------------------------------------------------------------------
<문제>
df에 있는 자료 중에서 짝수의 행 중에서 High, Low 컬럼만을 출력하는 코드를 작성하세요.

------------------------------------------------------------------
<요구사항1>
이전 강의에서 사용한 파일을 연속하여 실습하세요.

------------------------------------------------------------------
<요구사항2>
df.iloc[2,4,6,, ... ,:] 과 같은 코드는 안됩니다.

------------------------------------------------------------------
<요구사항3>
 [ 2 : 4: 2] 과 같은 슬라이싱 문법을 사용하면 안됩니다. 
 
------------------------------------------------------------------
<실행결과>

High	Low
0	797.659973	786.270020
2	796.676025	787.099976
4	792.739990	787.280029
6	794.229980	783.200012
8	782.780029	770.409973
...	...	...
52	833.409973	826.520020
54	842.000000	834.210022
56	848.684998	843.250000
58	848.630005	840.770020
60	853.400024	847.109985
31 rows × 2 columns


------------------------------------------------------------------
이상의 실습결과를 techmr@daum.net; ty@doublerock.io 메일로 전송하세요.
메일제목은 '013일차-실습-8-완료' 라고 지정하세요.

In [51]:
df[df.index % 2 == 0][['High', 'Low']]

High         Low
0   797.659973  786.270020
2   796.676025  787.099976
4   792.739990  787.280029
6   794.229980  783.200012
8   782.780029  770.409973
..         ...         ...
52  833.409973  826.520020
54  842.000000  834.210022
56  848.684998  843.250000
58  848.630005  840.770020
60  853.400024  847.109985

[31 rows x 2 columns]

In [52]:
df[0::2][['High', 'Low']]

High         Low
0   797.659973  786.270020
2   796.676025  787.099976
4   792.739990  787.280029
6   794.229980  783.200012
8   782.780029  770.409973
..         ...         ...
52  833.409973  826.520020
54  842.000000  834.210022
56  848.684998  843.250000
58  848.630005  840.770020
60  853.400024  847.109985

[31 rows x 2 columns]

In [53]:
print(df)

          Date        Open        High         Low       Close   Volume
0   12/19/2016  790.219971  797.659973  786.270020  794.200012  1225900
1   12/20/2016  796.760010  798.650024  793.270020  796.419983   925100
2   12/21/2016  795.840027  796.676025  787.099976  794.559998  1208700
3   12/22/2016  792.359985  793.320007  788.580017  791.260010   969100
4   12/23/2016  790.900024  792.739990  787.280029  789.909973   623400
..         ...         ...         ...         ...         ...      ...
56   3/13/2017  844.000000  848.684998  843.250000  845.539978  1149500
57   3/14/2017  843.640015  847.239990  840.799988  845.619995   779900
58   3/15/2017  847.590027  848.630005  840.770020  847.200012  1379600
59   3/16/2017  849.030029  850.849976  846.130005  848.780029   970400
60   3/17/2017  851.609985  853.400024  847.109985  852.119995  1712300

[61 rows x 6 columns]


In [54]:
print(df.Date[0])

12/19/2016
